Assets/Scripts/NPC/Grid.cs

```diff
@@ -0,0 +1,26 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Grid : MonoBehaviour
+{
+    [SerializeField] private List<Waypoint> waypoints = new List<Waypoint>();
+
+    public void SubscribeToGrid(Waypoint waypoint)
+    {
+        if (waypoints.Contains(waypoint)) return;
+        waypoints.Add(waypoint);
+    }
+
+    public void UnsubscribeFromGrid(Waypoint waypoint)
+    {
+        if (!waypoints.Contains(waypoint)) return;
+
+        waypoints.Remove(waypoint);
```

Comment:
Zelfde feedback als hier boven, de if kan op 1 line.
Maar
```suggestion
        if (!waypoints.Contains(waypoint)) return;
        waypoints.Remove(waypoint);
```